packages/portal-proto/src/features/cohortBuilder/CohortManager/CohortManager.tsx

```diff
@@ -0,0 +1,352 @@
+    deleteCohort,
+    coreDispatch,
+  ]);
+
+  const handleImport = useCallback(() => {
+    coreDispatch(showModal({ modal: Modals.ImportCohortModal }));
+  }, [coreDispatch]);
+
+  const handleAdd = useCallback(() => {
+    coreDispatch(addNewDefaultUnsavedCohort());
+  }, [coreDispatch]);
+
+  const updateCohortFilters = useCallback(
+    (field: string, operation: Operation) => {
+      coreDispatch(updateActiveCohortFilter({ field, operation }));
+    },
+    [coreDispatch],
+  );
+
+  return (
+    <div
+      data-tour="cohort_management_bar"
+      className="flex flex-row items-center justify-start gap-6 px-4 h-18 shadow-lg bg-primary"
+    >
+      {(isCohortIdFetching ||
+        isDeleteCohortLoading ||
+        isUpdateCohortLoading) && (
+        <LoadingOverlay data-testid="loading-spinner" visible />
+      )}
+
+      <div className="border-opacity-0">
+        <div className="flex flex-wrap gap-2 lg:gap-4">
+          <div className="flex justify-center items-center">
+            <Tooltip
+              label="Discard Unsaved Changes"
+              position="bottom"
+              withArrow
+            >
+              <span>
+                <CohortGroupButton
+                  data-testid="discardButton"
+                  onClick={() => setShowDiscard(true)}
+                  disabled={!cohortModified}
+                  $isDiscard={true}
+                  aria-label="Discard cohort changes"
+                >
+                  <DiscardIcon aria-hidden="true" />
+                </CohortGroupButton>
+              </span>
+            </Tooltip>
+
+            <CohortSelector
+              cohortId={cohortId}
+              cohortList={cohortList}
+              isSavedUnchanged={isSavedUnchanged}
+              cohortStatusMessage={cohortStatusMessage}
+              onCohortChange={handleCohortChange}
+            />
+          </div>
+          <CohortActions
+            onSave={() =>
+              currentCohort?.saved
+                ? setShowUpdateCohort(true)
+                : setShowSaveCohort(true)
+            }
+            onSaveAs={() => setShowSaveAsCohort(true)}
+            onAdd={handleAdd}
+            onDelete={() => setShowDelete(true)}
+            onImport={handleImport}
+            onExport={handleExport}
+            isModified={cohortModified}
+            isSaved={cohortSaved}
+            hasUnsavedCohorts={hasUnsavedCohorts}
+            isExporting={exportCohortPending}
+          />
+        </div>
+      </div>
+
+      <CohortModals
+        currentCohort={currentCohort}
+        cohortName={cohortName}
+        cohortId={cohortId}
+        filters={filters}
+        modal={modal}
+        showDelete={showDelete}
+        showDiscard={showDiscard}
+        showSaveCohort={showSaveCohort}
+        showSaveAsCohort={showSaveAsCohort}
+        showUpdateCohort={showUpdateCohort}
+        onHideModal={() => coreDispatch(hideModal())}
+        onSetShowDelete={setShowDelete}
```

Comment:
Can we move some of these function definitions into `CohortModals` instead of defining them in the parent and passing them in as props? This is a rather overwhelming number of props and I think both files would read cleaner if they were defined where they are used.

Reply:
I just updated it.

Reply:
Cool, looks great now 👍